crates/resolver/src/name_server/name_server_pool.rs

```diff
@@ -172,6 +177,19 @@ where
             //   this reorders the inner set
             ServerOrderingStrategy::QueryStatistics => conns.sort_unstable(),
             ServerOrderingStrategy::UserProvidedOrder => {}
+            ServerOrderingStrategy::RoundRobin => {
+                let num_concurrent_reqs = if opts.num_concurrent_reqs > 1 {
+                    opts.num_concurrent_reqs
+                } else {
+                    1
+                };
+                if num_concurrent_reqs < conns.len() {
+                    let index = NEXT_INDEX.fetch_add(num_concurrent_reqs, AtomicOrdering::SeqCst)
+                        % conns.len();
+                    let temp = conns.drain(0..index).collect::<Vec<_>>();
+                    conns.extend(temp);
+                }
```

Comment:
Suggest we could use `Vec::rotate_left()` here.

It feels like there's more complexity here than necessary, can you explain why all of this is needed?